src/ClientResolver.php

```diff
@@ -1115,15 +1116,33 @@ public static function _default_signing_region(array &$args)
             : $args['region'];
     }
 
+    public static function _apply_region($value, array &$args)
+    {
+        if (empty($value)) {
+            self::_missing_region($args);
+        }
+        $args['region'] = $value;
+    }
+
+    public static function _default_region(&$args)
+    {
+        return ConfigurationResolver::resolve('region', '', 'string');
+    }
+
     public static function _missing_region(array $args)
     {
         $service = isset($args['service']) ? $args['service'] : '';
```

Comment:
When will service ever be blank?  Is there a good use case for resolving a client without a service using this file?